internal/event-collector/ykclient/client.go

```diff
@@ -156,26 +151,20 @@ func (c *Client) loadCurrentPartitions() ([]*dao.PartitionInfo, error) {
 		}
 		partitions = append(partitions, pi...)
 	}
-	// Insert partitions into DB
-	for _, p := range partitions {
-		// Insert partition into DB
-		_, err := c.repo.Dbpool.Exec(context.Background(), "INSERT INTO partitions (id, cluster_id, name, capacity, used_capacity, utilization, total_nodes, applications, total_containers, state, last_state_transition_time) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)",
-			uuid.NewString(), p.ClusterID, p.Name, p.Capacity.Capacity, p.Capacity.UsedCapacity, p.Capacity.Utilization, p.TotalNodes, p.Applications, p.TotalContainers, p.State, p.LastStateTransitionTime)
-		if err != nil {
-			return nil, fmt.Errorf("could not insert partition into DB: %v", err)
-		}
+	if err = c.repo.UpsertPartitions(partitions); err != nil {
+		return nil, err
 	}
 	return partitions, nil
 }
 
-func (c *Client) loadCurrentQueues(partitions []*dao.PartitionInfo) (map[string][]*dao.PartitionQueueDAOInfo, error) {
-	queuesByPartition := map[string][]*dao.PartitionQueueDAOInfo{}
+func (c *Client) loadCurrentQueues(partitions []*dao.PartitionInfo) ([]*dao.PartitionQueueDAOInfo, error) {
+	queues := []*dao.PartitionQueueDAOInfo{}
```

Comment:
I figured out that each queue struct, `dao.PartitionQueueDAOInfo`, already has the partition name field. So, no need to group them by partition name.